compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/TypeChecker.java

```diff
@@ -619,10 +619,18 @@ public void visit(BLangListConstructorExpr listConstructor) {
             if (arrayType.state == BArrayState.OPEN_SEALED) {
                 arrayType.size = listConstructor.exprs.size();
                 arrayType.state = BArrayState.CLOSED_SEALED;
-            } else if (arrayType.state != BArrayState.UNSEALED && arrayType.size != listConstructor.exprs.size()) {
-                dlog.error(listConstructor.pos,
-                        DiagnosticCode.MISMATCHING_ARRAY_LITERAL_VALUES, arrayType.size, listConstructor.exprs.size());
-                return;
+            } else if ((arrayType.state != BArrayState.UNSEALED) && (arrayType.size != listConstructor.exprs.size())) {
```

Comment:
If we move this logic to a function, we can improve the readability of the code. WDYT?

Reply:
It will be like two functions ..
```if (isSealedArray(arrayType) && sizeMismatch(arrayType, listConstructor.exprs))```. Will this help ?